fcrepo-client-impl/src/main/java/org/fcrepo/client/impl/FedoraRepositoryImpl.java

```diff
@@ -315,8 +327,117 @@ public boolean isWritable() {
     }
 
     @Override
+    public String startTransaction() throws FedoraException {
+        final HttpPost post = httpHelper.createPostMethod(FCR_TX, null);
+        try {
+
+            final HttpResponse response = httpHelper.execute(post);
+            final String uri = post.getURI().toString();
+            final StatusLine status = response.getStatusLine();
+            final int statusCode = status.getStatusCode();
+
+            if (statusCode == SC_CREATED) {
+                final String txId = response.getFirstHeader("Location").getValue().substring(repositoryURL.length());
+                TX_ID.set(txId);
+                LOGGER.debug("Started transaction {}", txId);
+                return txId;
+            } else if (statusCode == SC_FORBIDDEN) {
+                LOGGER.error("request to start new transaction {} is not authorized.", uri);
+                throw new ForbiddenException("request to start new transaction " + uri + " is not authorized.");
+            } else {
+                LOGGER.error("error start new transaction {}: {} {}", uri, statusCode, status.getReasonPhrase());
+                throw new FedoraException("error start new transaction " + uri + ": " + statusCode + " " +
+                        status.getReasonPhrase());
+            }
+
+        } catch (final Exception e) {
+            LOGGER.error("could not encode URI parameter", e);
+            throw new FedoraException(e);
+        } finally {
+            post.releaseConnection();
+        }
+
+    }
+
+    @Override
+    public void commitTransaction() throws FedoraException {
+        final HttpPost post = httpHelper.createPostMethod(TX_ID.get()
+                + FCR_TX + FCR_COMMIT, null);
+        try {
+            final HttpResponse response = httpHelper.execute(post);
+            final String uri = post.getURI().toString();
+            final StatusLine status = response.getStatusLine();
+            final int statusCode = status.getStatusCode();
+
+            if (statusCode == SC_FORBIDDEN) {
+                LOGGER.error("request to commit transaction {} is not authorized.", uri);
+                throw new ForbiddenException("request to commit transaction " + uri + " is not authorized.");
+            } else {
+                if (statusCode != SC_NO_CONTENT) {
+                    LOGGER.error("error committing back transaction {}: {} {}", uri,
+                            statusCode, status.getReasonPhrase());
+                    throw new FedoraException("error committing transaction " + uri + ": " + statusCode + " " +
+                            status.getReasonPhrase());
+                }
+            }
+            LOGGER.debug("Committed transaction {}", TX_ID.get());
+        } catch (final Exception e) {
+            LOGGER.error("could not encode URI parameter", e);
```

Comment:
Please do not dump exception stacktraces into the log. Instead, print the `e.getMessage()`.